templates/CRM/Case/Page/Tab.tpl

```diff
@@ -35,17 +35,17 @@
     <div class="view-content">
     <div class="help">
          {ts 1=$displayName}This page lists all case records for %1.{/ts}
-         {if $permission EQ 'edit' and call_user_func(array('CRM_Core_Permission','check'), 'access all cases and activities') and $allowToAddNewCase}
+         {if $permission EQ 'edit' and $allowToAddNewCase}
+         {crmPermission has='access all cases and activities'}
          {ts 1="href='$newCaseURL' class='action-item'"}Click <a %1>Add Case</a> to add a case record for this contact.{/ts}{/if}
```

Comment:
The closing `{/if}` is at the end of the line here, but the corresponding `{/crmPermission}` should be here instead of way down at the bottom.